Gems/ROS2/Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,485 @@
+            auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+            AZ_Assert(meshGeometry, "geometry is not meshGeometry");
+
+            auto asset = PrefabMakerUtils::GetAssetFromPath(*m_urdfAssetsMapping, meshGeometry->filename);
+            if (!asset)
+            {
+                return;
+            }
+            // Get asset path for a given model path
+            const auto azMeshPath = AZ::IO::Path(asset->m_sourceAssetGlobalPath);
+
+            AZStd::optional<AZ::IO::Path> pxmodelPath = Internal::GetMeshProductPathFromSourcePath(azMeshPath);
+            if (!pxmodelPath)
+            {
+                AZ_Error(Internal::collidersMakerLoggingTag, false, "Could not find pxmodel for %s", azMeshPath.c_str());
+                entity->Deactivate();
+                return;
+            }
+
+            // Get asset product id (pxmesh)
+            AZ::Data::AssetId assetId;
+            AZ::Data::AssetType assetType = AZ::AzTypeInfo<PhysX::Pipeline::MeshAsset>::Uuid();
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                assetId, &AZ::Data::AssetCatalogRequests::GetAssetIdByPath, pxmodelPath->c_str(), assetType, false);
+            // Insert pxmesh into the collider component
+            PhysX::MeshColliderComponentRequestsBus::Event(entityId, &PhysX::MeshColliderComponentRequests::SetMeshAsset, assetId);
+            entity->Deactivate();
+
+            return;
+        }
+
+        AZ_Printf(Internal::collidersMakerLoggingTag, "URDF geometry type: %d\n", geometry->type);
+        switch (geometry->type)
+        {
+        case urdf::Geometry::SPHERE:
+            {
+                auto sphereGeometry = std::dynamic_pointer_cast<urdf::Sphere>(geometry);
+                AZ_Assert(sphereGeometry, "geometry is not sphereGeometry");
+                const Physics::SphereShapeConfiguration cfg{ static_cast<float>(sphereGeometry->radius) };
+                entity->CreateComponent<PhysX::EditorColliderComponent>(colliderConfig, cfg);
+            }
+            break;
+        case urdf::Geometry::BOX:
+            {
+                const auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                AZ_Assert(boxGeometry, "geometry is not boxGeometry");
+                const Physics::BoxShapeConfiguration cfg{ URDF::TypeConversions::ConvertVector3(boxGeometry->dim) };
+                entity->CreateComponent<PhysX::EditorColliderComponent>(colliderConfig, cfg);
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                AZ_Assert(cylinderGeometry, "geometry is not cylinderGeometry");
+                // TODO HACK Underlying API of O3DE  does not have Physic::CylinderShapeConfiguration implementation
```

Comment:
You can make a Github issue within the main O3DE repo for the SIG-Simulation group to implement this for the CylinderShadeConfiguration struct.